src/Integration/Helpers/RuleSetHelper.cs

```diff
@@ -95,9 +95,24 @@ public static RuleSetInclude FindInclude(RuleSet source, RuleSet target)
 
             string relativeTargetFilePath = PathHelper.CalculateRelativePath(source.FilePath, target.FilePath);
 
-            return source.RuleSetIncludes.SingleOrDefault(i =>
+            var matchingRuleSetIncludes = source.RuleSetIncludes.Where(i =>
                 StringComparer.OrdinalIgnoreCase.Equals(i.FilePath, relativeTargetFilePath)
-                || StringComparer.OrdinalIgnoreCase.Equals(i.FilePath, target.FilePath));
+                || StringComparer.OrdinalIgnoreCase.Equals(i.FilePath, target.FilePath))
+                .ToList();
+
+            if (matchingRuleSetIncludes.Count == 0)
+            {
+                return null;
+            }
+            else if (matchingRuleSetIncludes.Count == 1)
+            {
+                return matchingRuleSetIncludes[0];
+            }
+            else
+            {
+                Debug.Fail("Not expecting to find multiple RuleSetInclude matching the filter");
+                return null;
```

Comment:
It looks like this methods is only used in _SolutionBindingInformationProvider.IsFullyBoundProject_, and this change will cause _IsFullyBound_ to returning _false_, which will trigger the Error List gold bar.
I don't think is the behaviour we want; I don't think it matters in this case if the ruleset is included multiple times, although it would be worth updating the method comment to make it clear that this is the behaviour.
What do you think?

Reply:
So I keep the `Debug.Fail`, I add a comment and return the first element, does this sound OK?

Reply:
Looks good to me.